src/lib/datepicker/calendar.ts

```diff
@@ -29,7 +30,39 @@ import {MatDatepickerIntl} from './datepicker-intl';
 import {MatMonthView} from './month-view';
 import {MatMultiYearView, yearsPerPage} from './multi-year-view';
 import {MatYearView} from './year-view';
+import {ComponentPortal, ComponentType, Portal} from '@angular/cdk/portal';
+import {Subject} from 'rxjs/Subject';
+import {takeUntil} from 'rxjs/operators/takeUntil';
 
+/** Default header for MatCalendar */
+@Component({
+    moduleId: module.id,
+    selector: 'mat-calendar-header',
+    templateUrl: 'calendar-header.html',
+    encapsulation: ViewEncapsulation.None,
+    preserveWhitespaces: false,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MatCalendarHeader implements OnDestroy {
+  /** Subject that emits when the component has been destroyed. */
+  private _destroyed = new Subject<void>();
+
+  private _intlChanges: Subscription;
+
+  constructor(@Host() public calendar: MatCalendar<any>,
+              private _intl: MatDatepickerIntl,
+              changeDetectorRef: ChangeDetectorRef) {
+    this._intlChanges = _intl.changes.pipe(takeUntil(this._destroyed)).subscribe(
```

Comment:
```ts
intl.changes.pipe(takeUntil(this._destroyed)).subscribe(() => changeDetectorRef.markForCheck());
```

Reply:
@mmalerba It looks good now!

There is still a problem with two tests (build):
- https://circleci.com/gh/angular/material2/3888?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
- https://travis-ci.org/angular/material2/jobs/351039040

I think this is not related to this PR as I see builds failing in other PR too.

Reply:
Yeah those failures can just be ignored. So just a couple things left before we can submit:
- Need to clean up the `MatCalendar` API so it doesn't have methods and properties that really belong in the header
- Also move over any tests related to logic that we move into the header
- Get a green CLA status, which should just happen soon since your university has already signed the CLA